src/gui/guiFormSpecMenu.cpp

```diff
@@ -4452,6 +4452,12 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 				if (m_left_drag_stacks.size() > 1) {
 					// Finalize the left-dragging
 					for (auto &ds : m_left_drag_stacks) {
+						if (ds.first == *m_selected_item) {
+							// This entry is needed to properly calculate the stack sizes.
+							// The stack already exists, hence no further action needed here.
+							continue;
+						}
```

Comment:
`m_left_drag_stacks` contains the source stack and all the destination slots to divide the item count equally. Since we are moving items *from* the source stack, this specific movement action is not needed to be sent. It's an optimization and not related to the server-side bugfix.

Reply:
Hmm okay. Can you split this into a separate commit?

Reply:
Done. I found that this change also prevents crashes of older servers when a new client attempts to reproduce this bug. Since it does not fix the root of the problem it does make sense to have two commits.